web/src/message_fetch.ts

```diff
@@ -194,13 +243,22 @@ function handle_operators_supporting_id_based_api(data) {
         return data;
     }
 
-    const parsed_narrow_data = JSON.parse(data.narrow);
+    const parsed_narrow_data = z.array(narrow_term_schema).parse(JSON.parse(data.narrow));
```

Comment:
Does the `narrow_term_schema`'s `operand` field require a fix??

Reply:
Uhm, I don't think so. This function just modifies the `data.narrow` based on some conditions to help in other functions.